openpype/hosts/houdini/plugins/publish/extract_opengl.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+import pyblish.api
+import openpype.api
+from openpype.hosts.houdini.api.lib import render_rop
+
+import hou
+
+
+class ExtractOpenGL(openpype.api.Extractor):
+
+    order = pyblish.api.ExtractorOrder
+    label = "Extract OpenGL"
+    families = ["review"]
+    hosts = ["houdini"]
+    optional = True
+
+    def process(self, instance):
+
+        ropnode = hou.node(instance.data["instance_node"])
+
+        # Get the filename from the filename parameter
+        # `.evalParm(parameter)` will make sure all tokens are resolved
+        output = ropnode.evalParm("picture")
+        staging_dir = os.path.dirname(output)
+        instance.data["stagingDir"] = staging_dir
+        file_name = os.path.basename(output)
+
+        # We run the render
+        self.log.info("Extracting '%s' to '%s'" % (file_name, staging_dir))
+        render_rop(ropnode)
+
+        # Unfortunately user interrupting the extraction does not raise an
+        # error and thus still continues to the integrator. To capture that
+        # we make sure all files exist
+        files = instance.data["frames"]
+        missing = [fname for fname in files
```

Comment:
maybe raise the error as soon as you find if a file doesn't exist?

Reply:
The benefit of doing it like this is that we are able to report to the user all missing files so they can investigate. Otherwise they'd see the single file reported only.

Reply:
yeah makes sense